src/fairchem/core/common/relaxation/optimizers/lbfgs_torch.py

```diff
@@ -16,87 +16,70 @@
 import torch
 from torch_scatter import scatter
 
-from fairchem.core.common.relaxation.ase_utils import batch_to_atoms
-from fairchem.core.common.utils import radius_graph_pbc
-
 if TYPE_CHECKING:
-    from torch_geometric.data import Batch
+    from .optimizable import OptimizableBatch
 
 
 class LBFGS:
+    """Limited memory BFGS optimizer for batch ML relaxations."""
+
     def __init__(
         self,
-        batch: Batch,
-        model: TorchCalc,
+        optimizable_batch: OptimizableBatch,
         maxstep: float = 0.01,
         memory: int = 100,
         damping: float = 0.25,
         alpha: float = 100.0,
-        force_consistent=None,
         device: str = "cuda:0",
         save_full_traj: bool = True,
         traj_dir: Path | None = None,
-        traj_names=None,
-        early_stop_batch: bool = False,
+        traj_names: list[str] | None = None,
     ) -> None:
-        self.batch = batch
-        self.model = model
+        """
+        Args:
+            optimizable_batch: an optimizable batch which includes a model and a batch of data
+            maxstep: maximum number of steps to run optimization
+            memory: Number of steps to be stored in memory
+            damping: The calculated step is multiplied with this number before added to the positions.
+            alpha: Initial guess for the Hessian (curvature of energy surface)
+            device: device to run optimization on
+            save_full_traj: wether to save full trajectory
+            traj_dir: path to save trajectories in
+            traj_names: list of trajectory files names
+        """
+        self.optimizable = optimizable_batch
         self.maxstep = maxstep
         self.memory = memory
         self.damping = damping
         self.alpha = alpha
         self.H0 = 1.0 / self.alpha
-        self.force_consistent = force_consistent
         self.device = device
         self.save_full = save_full_traj
         self.traj_dir = traj_dir
         self.traj_names = traj_names
-        self.early_stop_batch = early_stop_batch
-        self.otf_graph = model.model._unwrapped_model.otf_graph
+        self.trajectories = None
+
+        self.fmax = None
+        self.steps = None
```

Comment:
Do we need to define these here? They get instantiated in `run()` Same with `self.trajectories`

Reply:
Just standard practice to have happy IDEs, and gives minor memory/performance boosts https://peps.python.org/pep-0412/